x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -6,98 +6,100 @@
+          },
+        });
       }
-
-      setTime({
-        ...getTime(),
-        ...dateRange,
-      });
-
-      queryString.setQuery({ ...queryString.getQuery(), ...query });
-      // Unified search holds the all state, we need to force the hook to rerender so that it can return the most recent values
-      // This can be removed once we get the state from the URL
-      forceUpdate();
     },
-    [filterManager, queryString, getTime, setTime]
+    [filterManager, getRangeInTimestamp, getTime, dispatch]
   );
 
+  // This won't prevent onSubmit from being fired when `clear filters` is clicked,
+  // that happens because both onQuerySubmit and onFiltersUpdated are internally triggered on same event by SearchBar.
+  // This just delays potential duplicate onSubmit calls
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debounceOnSubmit = useCallback(debounce(onSubmit, 100), [onSubmit]);
+
   const saveQuery = useCallback(
     (newSavedQuery: SavedQuery) => {
       const savedQueryFilters = newSavedQuery.attributes.filters ?? [];
       const globalFilters = filterManager.getGlobalFilters();
-      filterManager.setFilters([...savedQueryFilters, ...globalFilters]);
 
-      // Unified search holds the all state, we need to force the hook to rerender so that it can return the most recent values
-      // This can be removed once we get the state from the URL
-      forceUpdate();
+      const query = newSavedQuery.attributes.query;
+
+      dispatch({
+        type: 'setQuery',
+        payload: {
+          query,
+          filters: [...savedQueryFilters, ...globalFilters],
+        },
+      });
     },
-    [filterManager]
+    [filterManager, dispatch]
   );
 
   const clearSavedQUery = useCallback(() => {
-    filterManager.setFilters(filterManager.getGlobalFilters());
-
-    // Unified search holds the all state, we need to force the hook to rerender so that it can return the most recent values
-    // This can be removed once we get the state from the URL
-    forceUpdate();
-  }, [filterManager]);
+    dispatch({
+      type: 'setFilter',
+      payload: filterManager.getGlobalFilters(),
+    });
+  }, [filterManager, dispatch]);
 
   const buildQuery = useCallback(() => {
     if (!metricsDataView) {
       return null;
     }
-    return buildEsQuery(metricsDataView, queryString.getQuery(), filterManager.getFilters());
-  }, [filterManager, metricsDataView, queryString]);
+    return buildEsQuery(metricsDataView, state.query, state.filters);
+  }, [metricsDataView, state.filters, state.query]);
 
   return {
-    dateRangeTimestamp: currentTimeRange,
-    esQuery: buildQuery(),
-    submitFilterChange,
+    dateRangeTimestamp: state.dateRangeTimestamp,
+    buildQuery,
+    onSubmit: debounceOnSubmit,
     saveQuery,
     clearSavedQUery,
-    unifiedSearchQuery: queryString.getQuery() as Query,
+    // we'll use the hooks state instad of unified search's
```

Comment:
I guess we can remove this comment. Wdyt?

Reply:
Yep, removed ✅